internal/resources/metal/gateway/models.go

```diff
@@ -64,9 +18,14 @@ type DataSourceModel struct {
 	State                 types.String `tfsdk:"state"`
 }
 
-func (m *DataSourceModel) parse(gw *metalv1.FindMetalGatewayById200Response) diag.Diagnostics {
+type ResourceModel struct {
+	BaseModel
+	Timeouts timeouts.Value `tfsdk:"timeouts"`
+}
+
+func (m *BaseModel) parse(gw *metalv1.FindMetalGatewayById200Response) diag.Diagnostics {
```

Comment:
Will we generally need a `parse` method in composite types to pickup the fields that are not defined in the embedded model?

 `func (m *DataSourceModel) DataSourceModel parse(...) { m.BaseModel.parse(...); m.GatewayID = ... }`

Reply:
I see Organization parse following that pattern, where the DataSourceModel relies on the ResourceModel's parse before adding additional fields. It seems this pattern may not have been consistently applied.

Reply:
The `parse` method is for copying API response values into a model, so it's only concerned with API-defined properties.  In the case of this `m.Timeouts`, that is a Terraform-defined property that has no meaning to the API; the `m.Timeouts` value is always set by the plugin framework based on the `tfsdk: "timeouts"` tag.

In cases where a composite type has additional API-defined properties, you are correct that there needs to be a `parse` method attached to the composite type that delegates to the underlying `parse` method(s) and also sets the additional properties.  There's an example of that in the `metal_organization` resource/datasource.

Reply:
For reference: https://github.com/equinix/terraform-provider-equinix/pull/853/files#diff-c4f9c071e0fec3ac370badc0f9a8439beef2dffe2394809bda88960ed53a6a2aR54-R58